api/segments/models.py

```diff
@@ -181,9 +183,18 @@ def does_identity_match(
 
         for trait in traits:
             if trait.trait_key == self.property:
-                if trait.value_type == INTEGER:
+                if self.operator == MODULO:
+                    if trait.value_type in [INTEGER, FLOAT]:
```

Comment:
@matthewelwell Review the logic again. While writing test script I found that if a trait with `value_type = STRING|BOOLEAN` is  used along with `MODULO`, `does_identity_match` returns `None` instead of a `boolean` value. Have updated the logic accordingly.

Reply:
Great, thanks!

Reply:
I actually think that we'd be better off simply adding a default `return False` statement at the end of this method?